mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -352,61 +452,79 @@ def list_artifacts(self, path=None):
         return infos
 
     def _download_file(self, remote_file_path, local_path):
-        run_relative_remote_file_path = posixpath.join(
```

Comment:
In the interest of deduplicating credential resolution logic, I've centralized the logic for both file and directory downloads in `download_artifacts()` and left an explanatory comment in the now-empty-bodied `_download_file` function.